### PR TITLE
[DX] Reduce the use of direct persist/flush in Behat context, InstallerBundle and NotificationBundle

### PR DESCRIPTION
**Pull request by nidup**
Group any uses of persist/remove/flush of the Doctrine ObjectManager in SaverInterface/RemoverInterface to prepare a shiny future (be able to rely on business events).

TODO : 
 - [x] code review
 - [x] rebase

NB: encountered unstable that are fixed by https://github.com/akeneo/pim-community-dev/pull/4053/files

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | N
| Changelog updated                 | Y
| Review and 2 GTM                  | Y

### CODE REVIEW

File: features/Context/FixturesContext.php [hunk 1]
```diff
@@ -1226,8 +1226,8 @@ public function getUsername()
      */
     public function iVeRemovedTheAttribute($attribute)
```
Comment (juliensnz) [inline features/Context/FixturesContext.php]: not your fault but really ugly method name

File: CHANGELOG-1.6.md [hunk 2]
```diff
@@ -22,3 +22,4 @@
     `Pim\Component\Connector\ArrayConverter\Flat\VariantGroupStandardConverter` and
     `Pim\Component\Connector\ArrayConverter\Structured\AttributeOptionStandardConverter`
 - Remove deprecated argument $propertyCopier from constructor of `Pim\Component\Catalog\Updater\ProductUpdater` and allow to inject supported fields
+- Remove argument $em from constructor of `Pim\Bundle\NotificationBundle\Manager\NotificationManager` and inject SaverInterface and RemoverInterface
```
Comment (skeleton) [inline CHANGELOG-1.6.md]: add full namespace for SaverInterface & RemoverInterface

File: features/group-type/filter_group_types.feature [hunk 3]
```diff
@@ -6,14 +6,13 @@ Feature: Filter group types
...
     Given the "default" catalog configuration
-    And there is no group type
     And the following group types:
       | code    | label    |
       | related | Relation |
       | special | Special  |
     And I am logged in as "Peter"
     Given I am on the group types page
-    Then the grid should contain 2 elements
+    Then the grid should contain 4 elements
```
Comment (jjanvier) [inline features/group-type/filter_group_types.feature]: is it normal we change that?

File: src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml [hunk 4]
```diff
@@ -6,3 +6,19 @@ services:
             - '@akeneo_storage_utils.saver.base_options_resolver'
             - '@event_dispatcher'
             - %akeneo_batch.entity.job_instance.class%
+
+    akeneo_batch.saver.job_execution:
+        class: %akeneo_storage_utils.saver.base.class%
+        arguments:
+            - '@doctrine.orm.entity_manager'
```
Comment (jjanvier) [inline src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml]: I thought a different em was used by the batch
am I wrong?

File: features/group-type/filter_group_types.feature (hunk 3 above)
Comment (skeleton) [inline features/group-type/filter_group_types.feature]: why 4 ?

**Comment by jjanvier** [comment]
GTM if questions are OK

**Comment by juliensnz** [comment]
GTM, Awesome

**Comment by skeleton** [comment]
gtm once fixed

File: features/group-type/filter_group_types.feature (hunk 3 above)
Comment (nidup) [inline features/group-type/filter_group_types.feature]: cause there is already 2 in the catalog we load (i dropped the method which resets the group type table)

File: src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml (hunk 4 above)
Comment (nidup) [inline src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml]: you right i started to fix then see i have issue, i remove this service until i fix batch for real

File: features/Context/FixturesContext.php (hunk 1 above)
Comment (nidup) [inline features/Context/FixturesContext.php]: i change it

File: src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml (hunk 4 above)
Comment (BitOne) [inline src/Akeneo/Bundle/BatchBundle/Resources/config/savers.yml]: You should use the (ugly) JobRepository service provided by the batchBundle which is able to save and load jobs using the right em.

File: CHANGELOG-1.6.md (hunk 2 above)
Comment (nidup) [inline CHANGELOG-1.6.md]: done

**Pull request merged**